packages/fiori-annotation-api/src/cds/writer.ts

```diff
@@ -963,54 +990,10 @@ function printChange(parent: ContainerNode | undefined) {
     };
 }
 
-function indexInserts<T extends ElementInserts>(
-    changes: Array<T>,
-    containerSize: number
-): [(number | undefined)[], Map<number | undefined, T[]>] {
-    const changesByIndex = new Map<number | undefined, T[]>();
-    const indices: (number | undefined)[] = [];
-    for (const change of changes) {
-        const index =
-            change.index !== undefined && change.index > -1 ? Math.min(change.index, containerSize) : containerSize;
-        let changeSet = changesByIndex.get(index);
-        if (!changeSet) {
-            changeSet = [];
-            indices.push(index);
-            changesByIndex.set(index, changeSet);
-        }
-        changeSet.push(change);
-    }
-    return [indices, changesByIndex];
-}
-
-function findInsertPosition(
-    content: ContainerContentBlock[],
-    parent: ContainerNode,
-    index = -1
-): { position: Position; commaPosition?: Position } | undefined {
-    const childCount = getChildCount(parent);
-    if (childCount === 0) {
-        if (!parent.range) {
-            return undefined;
-        }
-        const position = copyPosition(parent.range.start);
-        position.character++; // range includes '{' or '[' characters
-        return { position };
-    }
-    const i = index > -1 ? Math.min(index, childCount) : childCount;
-    const { previousContentIndex, startContentIndex } = findContentIndices(content, i);
-    const anchor = getStartAnchor(content, parent, previousContentIndex, startContentIndex);
-    const previousElement = content[previousContentIndex];
-    if (anchor) {
-        if (previousElement?.type === 'element' && !previousElement.trailingComma && previousElement.element.range) {
-            return {
-                position: anchor,
-                commaPosition: copyPosition(previousElement.element.range.end)
-            };
-        }
-        return { position: anchor };
-    }
-    return undefined;
+function indexInserts<T extends ElementInserts>(change: T, containerSize: number): [number | undefined, T] {
+    const index =
```

Comment:
We probably can clean up this function even further. Only the number is currently used and change is passed through which isn't necessary. 

We can also consider removing the function all together, but maybe having a utility function for resolving index would be good to have.

Reply:
Changed.